### ISSUE TITLE
nan

### ISSUE THREAD

**Issue by ZhenboZhao77** [lwCVer/LWGANet#7]
[20250226_104409.log](https://github.com/user-attachments/files/18978030/20250226_104409.log)
请问作者，Epoch [4][650/6400]开始nan值这是为何？

**Comment by ZhenboZhao77** [comment]
_base_ = [
    '../_base_/datasets/dotav1.py', '../_base_/schedules/schedule_30e.py',
    '../_base_/default_runtime.py'
]

angle_version = 'le90'
\# fp16 = dict(loss_scale='dynamic')
model = dict(
    type='OrientedRCNN',
    backbone=dict(
        type='LWGANet',
        stem_dim=96,
        depths=(1, 4, 4, 2),
        att_kernel=(11, 11, 11, 11),
        norm_layer=dict(type='BN', requires_grad=True),
        fork_feat=True,
        drop_path_rate=0.1,
        init_cfg=dict(type='Pretrained',
checkpoint="/data/LWGANet/detection/lwganet_l2_e296.pth"),
        pretrained=None),
    neck=dict(
        type='FPN',
        in_channels=[96, 192, 384, 768],
        out_channels=256,
        num_outs=5),
    rpn_head=dict(
        type='OrientedRPNHead',
        in_channels=256,
        feat_channels=256,
        version=angle_version,
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8],
            ratios=[0.5, 1.0, 2.0],
            strides=[4, 8, 16, 32, 64]),
        bbox_coder=dict(
            type='MidpointOffsetCoder',
            angle_range=angle_version,
            target_means=[0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0, 0.5, 0.5]),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(
            type='SmoothL1Loss', beta=0.1111111111111111, loss_weight=1.0)),
    roi_head=dict(
        type='OrientedStandardRoIHead',
        bbox_roi_extractor=dict(
            type='RotatedSingleRoIExtractor',
            roi_layer=dict(
                type='RoIAlignRotated',
                out_size=7,
                sample_num=2,
                clockwise=True),
            out_channels=256,
            featmap_strides=[4, 8, 16, 32]),
        bbox_head=dict(
            type='RotatedShared2FCBBoxHead',
            in_channels=256,
            fc_out_channels=1024,
            roi_feat_size=7,
            num_classes=15,
            bbox_coder=dict(
                type='DeltaXYWHAOBBoxCoder',
                angle_range=angle_version,
                norm_factor=None,
                edge_swap=True,
                proj_xy=True,
                target_means=(.0, .0, .0, .0, .0),
                target_stds=(0.1, 0.1, 0.2, 0.2, 0.1)),
            reg_class_agnostic=True,
            loss_cls=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            loss_bbox=dict(type='SmoothL1Loss', beta=1.0, loss_weight=1.0))),
    train_cfg=dict(
        rpn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.7,
                neg_iou_thr=0.3,
                min_pos_iou=0.3,
                match_low_quality=True,
                gpu_assign_thr=800,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RandomSampler',
                num=256,
                pos_fraction=0.5,
                neg_pos_ub=-1,
                add_gt_as_proposals=False),
            allowed_border=0,
            pos_weight=-1,
            debug=False),
        rpn_proposal=dict(
            nms_pre=2000,
            max_per_img=2000,
            nms=dict(type='nms', iou_threshold=0.8),
            min_bbox_size=0),
        rcnn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.5,
                neg_iou_thr=0.5,
                min_pos_iou=0.5,
                match_low_quality=False,
                iou_calculator=dict(type='RBboxOverlaps2D'),
                gpu_assign_thr=800,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RRandomSampler',
                num=512,
                pos_fraction=0.25,
                neg_pos_ub=-1,
                add_gt_as_proposals=True),
            pos_weight=-1,
            debug=False)),
    test_cfg=dict(
        rpn=dict(
            nms_pre=2000,
            max_per_img=2000,
            nms=dict(type='nms', iou_threshold=0.8),
            min_bbox_size=0),
        rcnn=dict(
            nms_pre=2000,
            min_bbox_size=0,
            score_thr=0.05,
            nms=dict(iou_thr=0.1),
            max_per_img=2000)))

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='RResize', img_scale=(1024, 1024)),
    dict(
        type='RRandomFlip',
        flip_ratio=[0.25, 0.25, 0.25],
        direction=['horizontal', 'vertical', 'diagonal'],
        version=angle_version),
    dict(
        type='PolyRandomRotate',
        rotate_ratio=0.5,
        angles_range=180,
        auto_bound=False,
        rect_classes=[9, 11],
        version=angle_version),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]

data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(pipeline=train_pipeline, version=angle_version),
    val=dict(version=angle_version),
    test=dict(version=angle_version))

optimizer = dict(
    _delete_=True,
    type='AdamW',
    lr=0.0002,
    betas=(0.9, 0.999),
    weight_decay=0.05)

evaluation = dict(interval=10, metric='mAP', save_best='mAP')
runner = dict(type='EpochBasedRunner', max_epochs=30)
checkpoint_config = dict(interval=10, max_keep_ckpts=10)

**Comment by ZhenboZhao77** [comment]
这是配置文件，期待你的回复

**Comment by ZhenboZhao77** [comment]
或者说我只有一张显卡想要复现出效果需要更改学习率吗？还是应该做什么调整

**Comment by ZhenboZhao77** [comment]
OK  感谢



---- 回复的原邮件 ----
| 发件人 | Wei ***@***.***> |
| 日期 | 2025年02月26日 15:52 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [lwCVer/LWGANet] 训练损失为nan。 (Issue #7) |

或者说我只有一张显卡想要复现出效果需要更改学习率吗？还是应该做什么调整
感谢关注工作，我们是在4卡，即bs=2*4=8的情况下lr=0.0002，如果您只有单卡，建议尽量调大bs和训练epoch的同时逐步降低lr，

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

lwCVer left a comment (lwCVer/LWGANet#7)

或者说我只有一张显卡想要复现出效果需要更改学习率吗？还是应该做什么调整
感谢关注工作，我们是在4卡，即bs=2*4=8的情况下lr=0.0002，如果您只有单卡，建议尽量调大bs和训练epoch的同时逐步降低lr，
